### PR TITLE
feat(app): add delete confirm dialog

### PR DESCRIPTION
**Pull request by ThomasKro**
- add delete function for users
- add delete function for tenants
- add delete dialog for all delete functions
- fix onCascade Delete in schema
- add delete tenants permission

### CODE REVIEW

**Comment by ThomasKro** [comment]
Hi @beyer-johannes

File: packages/antify-admin/src/server/datasources/auth/schema.prisma [hunk 1]
```diff
@@ -61,11 +61,11 @@ model RolePermission {
...
 
 model UserTenantAccess {
-  user      User     @relation(fields: [userId], references: [id])
+  user      User     @relation(fields: [userId], references: [id], onDelete: Cascade)
   userId    String
-  tenant    Tenant   @relation(fields: [tenantId], references: [id])
+  tenant    Tenant   @relation(fields: [tenantId], references: [id], onDelete: Cascade)
   tenantId  String
-  role      Role     @relation(fields: [roleId], references: [id])
+  role      Role     @relation(fields: [roleId], references: [id], onDelete: Cascade)
```
Comment (beyer-johannes) [inline packages/antify-admin/src/server/datasources/auth/schema.prisma:68]: When a role get deleted, the users access to the tenant should not get deleted.

Solution 1: Do not allow to delete a role when one or more tenantAccesses exists.
Solution 2: On deleting a role, all associated tenantAccesses get replaced with a default role

**Review by beyer-johannes — CHANGES_REQUESTED** [review]

File: packages/antify-admin/src/server/datasources/auth/schema.prisma (hunk 1 above)
Comment (ThomasKro) [reply packages/antify-admin/src/server/datasources/auth/schema.prisma]: Fixed and added toaster if role is still assigned to a user.

File: packages/antify-admin/src/pages/admin/[tenantId]/roles/[roleId].vue [hunk 2]
```diff
@@ -30,10 +31,23 @@ const { data: permissions } = await useFetch(
...
-    method: 'DELETE',
-  });
+  const { data } = await useFetch<DeleteResponse>(
+    `/api/roles/${route.params.roleId}`,
+    {
+      ...useDefaultFetchOpts(),
+      method: 'DELETE',
+    }
+  );
+  console.log('error', data);
```
Comment (beyer-johannes) [inline packages/antify-admin/src/pages/admin/[tenantId]/roles/[roleId].vue:41]: found one

File: packages/antify-admin/src/pages/admin/[tenantId]/roles/[roleId].vue [hunk 3]
```diff
@@ -30,10 +31,23 @@ const { data: permissions } = await useFetch(
...
+
+  if (data.value && data.value.badRequest) {
+    console.log('error', data);
```
Comment (beyer-johannes) [inline packages/antify-admin/src/pages/admin/[tenantId]/roles/[roleId].vue:44]: and here

File: packages/antify-admin/src/server/api/roles/[roleId].delete.ts [hunk 4]
```diff
@@ -16,11 +17,19 @@ export default defineEventHandler(async (event) => {
     throw new HttpForbiddenError();
   }
 
-  await prisma.role.delete({
-    where: {
-      id: event.context.params.roleId
-    },
-  });
+  try {
```
Comment (beyer-johannes) [inline packages/antify-admin/src/server/api/roles/[roleId].delete.ts:20]: Please no try catch

**Review by beyer-johannes — CHANGES_REQUESTED** [review]

File: packages/antify-admin/src/glue/api/admin/[tenantId]/roles/[roleId].delete.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export type Response = {
+  badRequest?: {
+    message: String;
```
Comment (beyer-johannes) [inline packages/antify-admin/src/glue/api/admin/[tenantId]/roles/[roleId].delete.ts:3]: Please use errors: string[] and write the explicit error message into it

**Review by beyer-johannes — CHANGES_REQUESTED** [review]

File: packages/antify-admin/src/server/api/roles/[roleId].delete.ts [hunk 6]
```diff
@@ -16,9 +17,28 @@ export default defineEventHandler(async (event) => {
...
+      tenantId: true,
+      userId: true,
+      roleId: true,
+    },
+    where: {
+      roleId: event.context.params.roleId,
+    },
+  });
+
+  if (access && access.length > 0) {
```
Comment (beyer-johannes) [inline packages/antify-admin/src/server/api/roles/[roleId].delete.ts:31]: This query and condition is used on multiple places. Please outsorce into a service.

File: packages/antify-admin/src/server/api/roles/[roleId].get.ts [hunk 7]
```diff
@@ -19,22 +19,34 @@ export default defineEventHandler(async (event) => {
...
+      roleId: event.context.params.roleId,
+    },
+  });
+
   return {
     ...role,
-    permissions: role.permissions
-      .map(permission => permission.permissionId)
+    permissions: role.permissions.map((permission) => permission.permissionId),
+    canDelete: access.length === 0,
```
Comment (beyer-johannes) [inline packages/antify-admin/src/server/api/roles/[roleId].get.ts:50]: Outsource into a service

**Review by beyer-johannes — CHANGES_REQUESTED** [review]

**Review by beyer-johannes — APPROVED** [review]

**Pull request merged**